Code/Framework/AzCore/AzCore/Component/ComponentApplication.cpp

```diff
@@ -451,6 +453,9 @@ namespace AZ
             m_argV = &m_commandLineBufferAddress;
         }
 
+        // Check for a bootstrap config file that contains settings for configuring the allocator tracking settings
+        InitializeAllocatorSettings(m_argC, m_argV);
```

Comment:
Should we put this inside a #define so it can't be used in a shipped release build?

Reply:
No, because disabling this in release vs non-release build a decisions for us the engine developer to make, but a decision for the game developer to make.

Instead we can put this in a define that the game developer can change no-matter what the build configuration is, but defaults to a particular value based on release vs non-release configurations.

What I am saying is that doing a blanket #define check of `#if AZ_RELEASE_BUILD` build is over-indexing of making a decision that _we would make_, but not necessarily the game developer _wants to make_.
Instead we can add a define behind a CMake Cache variable ala the [ALLOW_SETTINGS_REGISTRY_DEVELOPMENT_OVERRIDES](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/CMakeLists.txt#L27) define that the game developer can toggle.

In code we would then put a function such as 
```
static constexpr bool GetAllowAllocatorInitializationFile()
{
#if defined(O3DE_ENABLE_ALLOCATOR_INITIALIZATION_STARTUP_FILE)
        return bool{ O3DE_ENABLE_ALLOCATOR_INITIALIZATION_STARTUP_FILE};
#elif AZ_RELEASE_BUILD
        return false;
#else
        return true;
#endif
}